lib/ttl_memoizeable.rb

```diff
@@ -44,9 +45,11 @@ def ttl_memoized_method(method_name, ttl: 1000)
       end
 
       define_method setup_memoization_method_name do
-        instance_variable_set(ttl_variable_name, expired_ttl) unless instance_variable_defined?(ttl_variable_name)
-        instance_variable_set(mutex_variable_name, Mutex.new) unless instance_variable_defined?(mutex_variable_name)
-        instance_variable_set(value_variable_name, nil) unless instance_variable_defined?(value_variable_name)
+        ::TTLMemoizeable::SetupMutex.synchronize do
```

Comment:
Since we call `setup_memoization_method_name` frequently, this will grab a global mutex each time, even when it isn't needed. Maybe we can return early when all the values are already set?

```suggestion
        return if instance_variable_defined?(ttl_variable_name) && instance_variable_defined?(mutex_variable_name) && instance_variable_defined?(value_variable_name)
        ::TTLMemoizeable::SetupMutex.synchronize do
```